Use existing Array/Date/Dictionary/URL wrappers for their CF types

beidson opened the pull request:
#### d3e0a4bcbe56c457cdcadbdf33037756f2b813f7
<pre>
Use existing Array/Date/Dictionary/URL wrappers for their CF types
<a href="https://bugs.webkit.org/show_bug.cgi?id=266288">https://bugs.webkit.org/show_bug.cgi?id=266288</a>
<a href="https://rdar.apple.com/119559705">rdar://119559705</a>

Reviewed by Alex Christensen.

* Source/WebKit/Platform/IPC/ArgumentCoders.h:
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.h:
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
(IPC::encodeSecureCodingDirectlyImpl):
(IPC::encodeObjectDirectly&lt;NSObject&lt;NSSecureCoding&gt;&gt;):
(IPC::decodeObjectDirectlyRequiringAllowedClasses&lt;NSObject&lt;NSSecureCoding&gt;&gt;):
(IPC::encodeObjectDirectly&lt;CFTypeRef&gt;):
* Source/WebKit/Shared/Cocoa/CoreIPCAVOutputContext.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCArray.h:
(WebKit::CoreIPCArray::createCFArray const):
* Source/WebKit/Shared/Cocoa/CoreIPCArray.mm:
(WebKit::CoreIPCArray::CoreIPCArray):
* Source/WebKit/Shared/Cocoa/CoreIPCArray.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCCFType.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCColor.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCContacts.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCDDActionContext.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCDDScannerResult.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCDate.h:
(WebKit::CoreIPCDate::createCFDate const):
(WebKit::CoreIPCDate::toID const):
(WebKit::CoreIPCDate::createDate const): Deleted.
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.h:
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.mm:
(WebKit::CoreIPCDictionary::CoreIPCDictionary):
(WebKit::CoreIPCDictionary::createCFDictionary const):
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCFont.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCLocale.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.mm:
(IPC::ArgumentCoder&lt;UniqueRef&lt;WebKit::ObjectValue&gt;&gt;::encode):
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCNSValue.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCPassKit.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCSecureCoding.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCString.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCURL.h:
(WebKit::CoreIPCURL::CoreIPCURL):
(WebKit::CoreIPCURL::createCFURL const):
* Source/WebKit/Shared/Cocoa/CoreIPCURL.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/cf/ArgumentCodersCF.cpp:
(IPC::ArgumentCoder&lt;CFArrayRef&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;RetainPtr&lt;CFArrayRef&gt;&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;CFDateRef&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;RetainPtr&lt;CFDateRef&gt;&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;CFDictionaryRef&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;RetainPtr&lt;CFDictionaryRef&gt;&gt;::decode): Deleted.
(IPC::ArgumentCoder&lt;CFURLRef&gt;::encode): Deleted.
(IPC::ArgumentCoder&lt;RetainPtr&lt;CFURLRef&gt;&gt;::decode): Deleted.
* Source/WebKit/Shared/cf/ArgumentCodersCF.h:
* Source/WebKit/Shared/cf/CFTypes.serialization.in:

Canonical link: <a href="https://commits.webkit.org/271948@main">https://commits.webkit.org/271948@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2a13018ee7008ee8c1446e4eb89ef7c1f2ebcaaf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30122 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8787 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31436 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32626 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27237 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30797 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10987 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/6032 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27252 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30422 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7379 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/25694 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6304 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/6461 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/26787 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33963 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/27472 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27220 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32640 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/6401 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4571 "Found 1 new test failure: fullscreen/full-screen-layer-dump.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30443 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8150 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7139 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7154 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6929 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/Shared/Cocoa/CoreIPCDDScannerResult.serialization.in:
```diff
@@ -22,7 +22,7 @@
 #if ENABLE(DATA_DETECTION)
 secure_coding_header: <pal/cocoa/DataDetectorsCoreSoftLink.h>
-[WebKitSecureCodingClass=PAL::getDDScannerResultClass()] webkit_secure_coding DDScannerResult {
+[WebKitSecureCodingClass=PAL::getDDScannerResultClass(), AdditionalEncoder=StreamConnectionEncoder] webkit_secure_coding DDScannerResult {
```
I'm a little surprised a DDScannerResult is sent via a StreamConnectionEncoder.  Is this really needed?  Same with DDActionContext, CNPhoneNumber, and CNPostalAddress.

beidson replied:
The way dictionary and array work, every type needs to have the same capabilities.
Bring in one, bring in all.

Perhaps we'll find time to improve that later!

beidson committed:
Use existing Array/Date/Dictionary/URL wrappers for their CF types
https://bugs.webkit.org/show_bug.cgi?id=266288
rdar://119559705

Reviewed by Alex Christensen.

* Source/WebKit/Platform/IPC/ArgumentCoders.h:
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.h:
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
(IPC::encodeSecureCodingDirectlyImpl):
(IPC::encodeObjectDirectly<NSObject<NSSecureCoding>>):
(IPC::decodeObjectDirectlyRequiringAllowedClasses<NSObject<NSSecureCoding>>):
(IPC::encodeObjectDirectly<CFTypeRef>):
* Source/WebKit/Shared/Cocoa/CoreIPCAVOutputContext.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCArray.h:
(WebKit::CoreIPCArray::createCFArray const):
* Source/WebKit/Shared/Cocoa/CoreIPCArray.mm:
(WebKit::CoreIPCArray::CoreIPCArray):
* Source/WebKit/Shared/Cocoa/CoreIPCArray.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCCFType.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCColor.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCContacts.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCDDActionContext.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCDDScannerResult.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCDate.h:
(WebKit::CoreIPCDate::createCFDate const):
(WebKit::CoreIPCDate::toID const):
(WebKit::CoreIPCDate::createDate const): Deleted.
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.h:
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.mm:
(WebKit::CoreIPCDictionary::CoreIPCDictionary):
(WebKit::CoreIPCDictionary::createCFDictionary const):
* Source/WebKit/Shared/Cocoa/CoreIPCDictionary.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCFont.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCLocale.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.mm:
(IPC::ArgumentCoder<UniqueRef<WebKit::ObjectValue>>::encode):
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCNSValue.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCPassKit.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCSecureCoding.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCString.serialization.in:
* Source/WebKit/Shared/Cocoa/CoreIPCURL.h:
(WebKit::CoreIPCURL::CoreIPCURL):
(WebKit::CoreIPCURL::createCFURL const):
* Source/WebKit/Shared/Cocoa/CoreIPCURL.serialization.in:
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/cf/ArgumentCodersCF.cpp:
(IPC::ArgumentCoder<CFArrayRef>::encode): Deleted.
(IPC::ArgumentCoder<RetainPtr<CFArrayRef>>::decode): Deleted.
(IPC::ArgumentCoder<CFDateRef>::encode): Deleted.
(IPC::ArgumentCoder<RetainPtr<CFDateRef>>::decode): Deleted.
(IPC::ArgumentCoder<CFDictionaryRef>::encode): Deleted.
(IPC::ArgumentCoder<RetainPtr<CFDictionaryRef>>::decode): Deleted.
(IPC::ArgumentCoder<CFURLRef>::encode): Deleted.
(IPC::ArgumentCoder<RetainPtr<CFURLRef>>::decode): Deleted.
* Source/WebKit/Shared/cf/ArgumentCodersCF.h:
* Source/WebKit/Shared/cf/CFTypes.serialization.in:

Canonical link: https://commits.webkit.org/271948@main
